### ISSUE TITLE
HTTP IncomingMessage constructor not recognized 

### ISSUE THREAD

**Issue by damianoviscardi** [jxcore/jxcore#681]
The following statement is false:
```javascript
req.constructor.name === 'IncomingMessage'
```
where req is an http.ClientRequest.

My issue is that I use an express-based MVC module that performs this check:
```javascript
if (!req || req.constructor.name !== 'IncomingMessage') {
		throw new Error('View Error: Express request object is required.');
	}
```
and throws error at every http request

**Comment by ktrzeciaknubisa** [comment]
Right, I see it with the following sample:

```js
var express = require('express');
var app = express();

app.get('/', function (req, res) {
  var n = req.constructor.name;
  if (!n) n = (n === "" ? "empty string" : typeof n);
  res.send('req.constructor.name = ' + n);
});

var server = app.listen(3000, function () {
  console.log('Example app listening at http://%s:%s', server.address().address, 3000);
});
```

**Comment by ktrzeciaknubisa** [comment]
I mean, when you run it with node, `req.constructor.name` == `IncomingMessage`, but with jx it is an empty string.

**Comment by damianoviscardi** [comment]
Exactly, to be compliant with native HTTP module, ```req.constructor.name``` should return ```'IncomingMessage'```

**Comment by ktrzeciaknubisa** [comment]
Ok, this gets simplier. It's not really related with express. As before, node and jx displays different results:

```js
var http = require('http');

var req = {
  __proto__: http.IncomingMessage.prototype
};

console.log(req.constructor.name);
```

**Comment by ktrzeciaknubisa** [comment]
I confirm that it's fixed as part of 0.3.1.0. Thanks @damianoviscardi
